src/test/java/se/sundsvall/businessrules/service/util/RuleEngineUtilTest.java

```diff
@@ -1,27 +1,109 @@
 package se.sundsvall.businessrules.service.util;
 
+import static java.util.UUID.randomUUID;
 import static org.assertj.core.api.Assertions.assertThat;
 
-import java.util.UUID;
 import java.util.stream.Stream;
 
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
+import se.sundsvall.businessrules.api.model.Fact;
+
 class RuleEngineUtilTest {
 
 	@ParameterizedTest
-	@MethodSource("isValidUuidArguments")
-	void isValidUuid(String uuid, boolean expectedResult) {
-		assertThat(RuleEngineUtil.isValidUuid(uuid)).isEqualTo(expectedResult);
+	@MethodSource("isValidUUIDArguments")
+	void isValidUUID(String uuid, boolean expectedResult) {
+		assertThat(RuleEngineUtil.isValidUUID(uuid)).isEqualTo(expectedResult);
 	}
 
-	private static Stream<Arguments> isValidUuidArguments() {
+	private static Stream<Arguments> isValidUUIDArguments() {
 		return Stream.of(
-			Arguments.of(UUID.randomUUID().toString(), true),
+			Arguments.of(randomUUID().toString(), true),
 			Arguments.of("not-valid-uuid", false),
 			Arguments.of("", false),
 			Arguments.of(null, false));
 	}
+
+	@ParameterizedTest
+	@MethodSource("hasValidUUIDArguments")
+	void hasValidUUIDValue(Fact fact, boolean expectedResult) {
+		assertThat(RuleEngineUtil.hasValidUUIDValue(fact)).isEqualTo(expectedResult);
+	}
+
+	private static Stream<Arguments> hasValidUUIDArguments() {
+		return Stream.of(
+			Arguments.of(Fact.create().withValue(randomUUID().toString()), true),
+			Arguments.of(Fact.create().withValue("not-valid-uuid"), false),
+			Arguments.of(Fact.create().withValue(""), false),
+			Arguments.of(Fact.create(), false));
+	}
+
+	@ParameterizedTest
+	@MethodSource("hasValidBooleanValueArguments")
+	void hasValidBooleanValue(Fact fact, boolean expectedResult) {
+		assertThat(RuleEngineUtil.hasValidBooleanValue(fact)).isEqualTo(expectedResult);
+	}
+
+	private static Stream<Arguments> hasValidBooleanValueArguments() {
+		return Stream.of(
+			Arguments.of(Fact.create().withValue("true"), true),
+			Arguments.of(Fact.create().withValue("TRUE"), true),
+			Arguments.of(Fact.create().withValue("false"), true),
+			Arguments.of(Fact.create().withValue("false"), true),
```

Comment:
Change "false" to "FALSE" (or remove, as this row is duplicate of row 55)?

Reply:
Yes, "FALSE" was my original intention here. But I failed to make my thoughts come true.